cli/src/lockfiles/python.rs

```diff
@@ -169,4 +225,25 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn parse_poetry_lock() {
+        let parser = Poetry::new(Path::new("tests/fixtures/poetry.lock")).unwrap();
+
+        let pkgs = parser.parse().unwrap();
+        assert_eq!(pkgs.len(), 62);
+
+        for pkg in &pkgs {
+            if pkg.name == "toml" {
+                assert_eq!(pkg.version, "0.10.2");
+                assert_eq!(pkg.package_type, PackageType::PyPi);
+            } else if pkg.name == "certifi" {
+                assert_eq!(pkg.version, "2021.10.8");
+                assert_eq!(pkg.package_type, PackageType::PyPi);
+            } else if pkg.name == "html5lib" {
+                assert_eq!(pkg.version, "1.1");
+                assert_eq!(pkg.package_type, PackageType::PyPi);
+            }
+        }
```

Comment:
The `if` inside the loop scares me, because it allows completely skipping the asserts. Imagine if the parser accidentally failed to set `name`: Then we'd iterate through all the packages and never check anything because `pkg.name` is never set to `"toml"`, `"certifi"`, or `"html5lib"`.

Using something like `.find(...).expect()` to ensure 

```suggestion
        let toml = pkgs
            .iter()
            .find(|p| p.name == "toml")
            .expect("Package 'toml' not found");
        assert_eq!(toml.version, "0.10.2");
        assert_eq!(toml.package_type, PackageType::PyPi);

        let certifi = pkgs
            .iter()
            .find(|p| p.name == "certifi")
            .expect("Package 'certifi' not found");
        assert_eq!(certifi.version, "2021.10.8");
        assert_eq!(certifi.package_type, PackageType::PyPi);

        let html5lib = pkgs
            .iter()
            .find(|p| p.name == "html5lib")
            .expect("Package 'html5lib' not found");
        assert_eq!(html5lib.version, "1.1");
        assert_eq!(html5lib.package_type, PackageType::PyPi);
```

It looks like you just copied the style of tests from `parse_pipfile()` and `lock_parse_pipfile()` above. They have the same problem...

You could extract that pattern into an `assert_package(...)` function if you want to fix all of them. And then use something like this in all of those tests:
```
assert_package(pkgs, "toml", "0.10.3", PackageType::PyPi);
```

Reply:
As you've mentioned, I just copied the existing tests. But I agree with you that it's too easy to pass these incorrectly. I'll add a function and fix the other tests too.